src/app/shared/code-mirror/code-mirror.component.ts

```diff
@@ -153,6 +175,39 @@ export class Class implements AfterContentInit, OnChanges, OnDestroy {
 		if( noCursor ) this.codeMirror.setOption( "readOnly", "nocursor" );
 		else this.setReadOnly( this.readOnly );
 	}
+
+	private validateQuery( currentString:string ) {
+		this.clearTextMarker();
+		try {
+			this.parser.parse( currentString );
+		} catch( error ) {
+			if( "message" in error && error.message.startsWith( "Parse error" ) ) {
+				this.displayParseError( error );
+			} else {
+				console.error( "Unexpected error while parsing the query", error );
+			}
+		}
+	}
+
+	private displayParseError( error ) {
+		let start = {
+			line: error.hash.loc.first_line - 1,
+			ch: error.hash.loc.first_column - 1
+		};
+		let end = {
+			line: error.hash.loc.last_line - 1,
+			ch: error.hash.loc.last_column + 1
+		};
+		// @ts-ignore
```

Comment:
Instead of ignoring type errors, find their origin and fix them (add missing interfaces, install missing `@types`, etc.). That way when reviewing the code again, you won't need to check CodeMirror's documentation (for example), and other developers will now what types are expected.